app/templating/AssetHelper.scala

```diff
@@ -86,6 +87,28 @@ trait AssetHelper { self: I18nHelper =>
       s"""<script src="$local"></script>"""
   }
 
-  def embedJsUnsafe(js: String): Html = Html(s"""<script>$js</script>""")
-  def embedJs(js: Html): Html = embedJsUnsafe(js.body)
+  def basicCsp(implicit req: RequestHeader): ContentSecurityPolicy = {
+    val assets = if (req.secure) "https://" + assetDomain else assetDomain
+    val socket = (if (req.secure) "wss://" else "ws://") + socketDomain
+    ContentSecurityPolicy(
+      defaultSrc = List("'self'", assets),
+      connectSrc = List("'self'", assets, socket, socket + ":*", lila.api.Env.current.ExplorerEndpoint, lila.api.Env.current.TablebaseEndpoint),
+      styleSrc = List("'self'", "'unsafe-inline'", assets, "https://fonts.googleapis.com"),
+      fontSrc = List("'self'", assetDomain, "https://fonts.gstatic.com"),
+      childSrc = List("'self'", assets, "https://www.youtube.com"),
+      imgSrc = List("data:", "*"),
+      scriptSrc = List("'self'", assets, "https://cdnjs.cloudflare.com")
+    )
+  }
+
+  def defaultCsp(implicit ctx: Context): ContentSecurityPolicy = {
+    implicit val req = ctx.req
+    basicCsp.withNonce(ctx.nonce)
```

Comment:
no need for `implicit val req = ctx.req` you can just `basicCsp(ctx.req).withNonce(ctx.nonce)`

implicit arguments are just arguments.